scripts/mail-scripts/main.go

```diff
@@ -12,64 +18,136 @@ import (
 	"gopkg.in/gomail.v2"
 )
 
+// Struct to store subscribers list array
 type subscriberList struct {
 	Subscribers []string `json:"subscribers"`
 }
 
 func main() {
+
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+
+	// Init goldmark configuration
 	markdown := goldmark.New(
 		goldmark.WithExtensions(
 			meta.Meta,
 		),
 	)
 
-	jsonFile, err := os.Open("emails.json")
-
+	// Open emails file for reading emails (Will be replaced in future)
+	emailFile, err := os.Open("emails.json")
 	if err != nil {
 		log.Print(err)
 	}
+	defer emailFile.Close()
 
-	defer jsonFile.Close()
+	// Read emails from the file into a buffer
+	byteValue, _ := ioutil.ReadAll(emailFile)
 
-	byteValue, _ := ioutil.ReadAll(jsonFile)
+	// Transferring email data to a var
+	var subsList subscriberList
+	json.Unmarshal(byteValue, &subsList)
 
-	var list subscriberList
+	// Reading content to be sent
+	// TODO: Replace this with logic to read file containing the latest post
+	content, _ := ioutil.ReadFile("../../content/post/example.md")
 
-	json.Unmarshal(byteValue, &list)
+	// Convert the markdown into a compatible HTML format
+	var htmlContent bytes.Buffer
+	if err := markdown.Convert(content, &htmlContent); err != nil {
+		log.Fatalln(err)
+	}
+	// Convert to string for using and tweaking it everywhere
+	contentString := htmlContent.String()
 
-	content, _ := ioutil.ReadFile("../../content/post/example.md")
+	// Iterate over emails list and start sending
+	for _, email := range subsList.Subscribers {
+
+		// Get the unsubscribe hash string
+		completeContent := addUnsubscribeLink(contentString, email)
+
+		// TODO here: Add image to the top of content if needed
+
+		// Send the email to the recipient
+		err := send(completeContent, email)
+		if err != nil {
+			log.Println("Error while sending mail to subscriber", email, "\nError : ", err)
+		}
+	}
+
+}
+
+// Utility to get the unsubscribe hash
+func addUnsubscribeLink(contentString string, email string) string {
 
-	var buf bytes.Buffer
+	// Get the encrypted hash to be sent
+	unsubString := encryptUnsubscribeString(email)
 
-	if err := markdown.Convert(content, &buf); err != nil {
-		panic(err)
+	// Prepare the HTML template of the unsubscribe option (raw as of now)
+	unSubscribeTemplate := fmt.Sprintf("<a href=\"/unsubscribe?uniqhash=%s\">UnSubscribe</a>", unsubString)
```

Comment:
```suggestion
	unSubscribeTemplate := fmt.Sprintf("<a href=\"%s/unsubscribe?uniqhash=%s\">UnSubscribe</a>", domain, unsubString)
```

We should have our production domain here, prolly should use an environment variable?

Reply:
Addressed in a11b7d5